### PR TITLE
Fix fake segment is never handled

### LINKED ISSUE #20934: [Bug]: Fake(empty) segment is never handled
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:master
- Deployment mode(standalone or cluster):
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

The empty segment is marked as fake, but the flush event from DataCoord only contains the segment ID, so IndexCoord and QueryCoord will nerver handle it

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Milvus Log

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by yah01**
Signed-off-by: yah01 <yang.cen@zilliz.com>
/kind bug
fix #20934

### CODE REVIEW

File: internal/metastore/kv/datacoord/kv_catalog.go [hunk 1]
```diff
@@ -169,7 +169,10 @@ func (kc *Catalog) AlterSegment(ctx context.Context, newSegment *datapb.SegmentI
 	maps.Copy(kvs, segmentKvs)
 	if newSegment.State == commonpb.SegmentState_Flushed && oldSegment.State != commonpb.SegmentState_Flushed {
 		flushSegKey := buildFlushedSegmentPath(newSegment.GetCollectionID(), newSegment.GetPartitionID(), newSegment.GetID())
-		newSeg := &datapb.SegmentInfo{ID: newSegment.GetID()}
+		newSeg := &datapb.SegmentInfo{
+			ID:     newSegment.GetID(),
+			IsFake: newSegment.GetNumOfRows() == 0,
```
Comment (jaime0815) [inline internal/metastore/kv/datacoord/kv_catalog.go:174]: The faked segment will only save into handleoff meta, It should not be saved here.

Comment (yah01) [reply internal/metastore/kv/datacoord/kv_catalog.go:174]: `newSeg` is the `SegmentInfo` we write flush event, will not be saved. `newSegment` is which will be saved

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**